File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    const result = [];
    const topK = [];
    
    for (let i = 0; i < score.length; i++) {
        topK.push(score[i]);
        topK.sort((a, b) => b - a);
        if (topK.length > k) {
            topK.pop();
        }
        result.push(topK.at(-1));
    }
    return result;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    result = []
    top_k = []
    
    for i in range(len(score)):
        top_k.append(score[i])
        top_k.sort(reverse=True)
        if len(top_k) > k:
            top_k.pop()
        result.append(top_k[-1])
    return result
            